src/libs/fileDownload/FileUtils.js

```diff
@@ -0,0 +1,98 @@
+import {Alert, Linking} from 'react-native';
+import moment from 'moment';
+import * as mime from 'react-native-mime-types';
+import CONST from '../../CONST';
+import * as Localize from '../Localize';
+
+/**
+ * Show alert on successful attachment download
+ */
+function showSuccessAlert() {
+    Alert.alert(
+        Localize.translateLocal('fileDownload.success.title'),
+        Localize.translateLocal('fileDownload.success.message'),
+        [
+            {
+                text: Localize.translateLocal('fileDownload.success.buttonOkText'),
+                style: 'cancel',
+            },
+        ],
+        {cancelable: false},
+    );
+}
+
+/**
+ * Show alert on attachment download error
+ */
+function showGeneralErrorAlert() {
+    Alert.alert(
+        Localize.translateLocal('fileDownload.generalError.title'),
+        Localize.translateLocal('fileDownload.generalError.message'),
+        [
+            {
+                text: Localize.translateLocal('fileDownload.generalError.buttonCancelText'),
+                style: 'cancel',
+            },
+        ],
+    );
+}
+
+/**
+ * Show alert on attachment download permissions error
+ */
+function showPermissionErrorAlert() {
+    Alert.alert(
+        Localize.translateLocal('fileDownload.permissionError.title'),
+        Localize.translateLocal('fileDownload.permissionError.message'),
+        [
+            {
+                text: Localize.translateLocal('fileDownload.permissionError.buttonCancelText'),
+                style: 'cancel',
+            },
+            {
+                text: Localize.translateLocal('fileDownload.permissionError.buttonSettingsText'),
+                onPress: () => Linking.openSettings(),
+            },
+        ],
+    );
+}
+
+/**
+ * Generating a random file name with timestamp and file extention
+ * @param {String} url
+ * @returns {String}
+ */
+function getAttachmentName(url) {
+    if (!url) {
+        return '';
+    }
+    return `${moment().format('DDMMYYYYHHmmss')}.${url.split(/[#?]/)[0].split('.').pop().trim()}`;
+}
+
+/**
+ * Returns file type based on the uri
+ * @param {String} fileUrl
+ * @returns {String}
+ */
+function getFileType(fileUrl) {
+    if (!fileUrl) {
+        return;
+    }
+    const fileName = fileUrl.split('/').pop().split('?')[0].split('#')[0];
+    const contentType = mime.contentType(fileName);
```

Comment:
I think it can be taken from the uploader thus thus function will be not-needed

Reply:
Do you mean from the backend?

Reply:
Oh, this is the download :headstone: But still there are many ways. 
1. Using Magic bytes.
2. Reading Response headers.
3. Reading file extensions. If you want this, then I think better would be extract the Code only for image and video from the lib into a regex.

Reply:
Well, 1 and 2 will still require downloading the file, and hence I didn't choose them. 

> better would be extract the Code only for image and video from the lib into a regex.

Didn't click that we could pull the dataset from the lib for regex. Thanks for the suggestion. I feel this is apt. What's your thought?

Reply:
Sorry, what is apt?

Reply:
Oh, I assume `appropriate`. I am fine with it.

Reply:
Updated